messaging/src/main/java/org/axonframework/queryhandling/interceptors/InterceptingQueryBus.java

```diff
@@ -16,108 +16,251 @@
+    }
+
+    @Nonnull
+    @Override
+    public MessageStream<QueryResponseMessage> subscriptionQuery(@Nonnull SubscriptionQueryMessage query,
+                                                                 @Nullable ProcessingContext context,
+                                                                 int updateBufferSize) {
+        // For subscription queries, delegate directly to the underlying bus
+        // The initial query and updates will be handled by the delegate bus
+        // Handler interceptors are already applied via subscribe(), so they will be invoked
+        // when the query is handled
+        return delegate.subscriptionQuery(query, context, updateBufferSize);
+    }
+
+    @Nonnull
+    @Override
+    public MessageStream<SubscriptionQueryUpdateMessage> subscribeToUpdates(@Nonnull SubscriptionQueryMessage query,
+                                                                            int updateBufferSize) {
+        return delegate.subscribeToUpdates(query, updateBufferSize);
+    }
+
+    @Nonnull
+    @Override
+    public CompletableFuture<Void> emitUpdate(@Nonnull Predicate<SubscriptionQueryMessage> filter,
+                                              @Nonnull Supplier<SubscriptionQueryUpdateMessage> updateSupplier,
+                                              @Nullable ProcessingContext context) {
+        if (updateDispatchInterceptors.isEmpty()) {
+            return delegate.emitUpdate(filter, updateSupplier, context);
+        }
+
+        try {
+            SubscriptionQueryUpdateMessage update = updateSupplier.get();
+            SubscriptionQueryUpdateMessage intercepted = interceptingResponseUpdateDispatcher.intercept(update, context);
+            return delegate.emitUpdate(filter, () -> intercepted, context);
+        } catch (Exception e) {
+            return CompletableFuture.failedFuture(e);
+        }
+    }
+
+    @Nonnull
+    @Override
+    public CompletableFuture<Void> completeSubscriptions(@Nonnull Predicate<SubscriptionQueryMessage> filter,
+                                                         @Nullable ProcessingContext context) {
+        return delegate.completeSubscriptions(filter, context);
+    }
+
+    @Nonnull
+    @Override
+    public CompletableFuture<Void> completeSubscriptionsExceptionally(
+            @Nonnull Predicate<SubscriptionQueryMessage> filter,
+            @Nonnull Throwable cause,
+            @Nullable ProcessingContext context
+    ) {
+        return delegate.completeSubscriptionsExceptionally(filter, cause, context);
+    }
+
+    private MessageStream<?> dispatchQuery(@Nonnull Message message,
+                                          @Nullable ProcessingContext processingContext) {
+        if (!(message instanceof QueryMessage query)) {
+            // The compiler should avoid this from happening.
+            throw new IllegalArgumentException("Unsupported message implementation: " + message);
+        }
```

Comment:
If this can't happen, why not just do an unconditional cast?  If it can happen, then I don't understand the comment.